### ISSUE TITLE
Canada.ca theme: Issues with Search in Top Nav

### ISSUE THREAD

**Issue by ciclerou** [wet-boew/GCWeb#1103]
The label's class "wb-inv" - visible = false fails technique H44 as per provided test case.  this is WCAG Level A.
3.3.2 Labels or Instructions: Labels or instructions are provided when content requires user input. (Level A) .

Test https://www.w3.org/TR/WCAG20-TECHS/H44.html
 For all input elements of type text, file or password, for all textareas and for all select elements in the Web page: 
 1.Check that there is a label element that identifies the purpose of the control before the input, textarea, or select element
 2.Check that the for attribute of the label element matches the id of the input, textarea, or select element
 3.Check that the label element is visible. 

Results
 1.1.1 - Checks #1 and #2 are true. (PASSES)
 3.3.2 - Check #3 is false (FAILS)

![searchcanada_source](https://cloud.githubusercontent.com/assets/2150730/14460770/4c626d46-008d-11e6-8726-7a2a88e64f2e.PNG)
![searchcanada_gui](https://cloud.githubusercontent.com/assets/2150730/14460773/54ce6a48-008d-11e6-81fc-1c825b017bc7.PNG)

**Comment by LaurentGoderre** [comment]
@ciclerou while it fails this specific test, WCAG clearly says that the criterion may be met in another way. Given that the hidden label is programatically associated and that a visual queue is provided (but not relied on) ion the form of the placeholder, this does not constitute a failure.

Now with that being said. It is possible that using the `title` attribute might be slightly better but that would mean a markup change.

**Comment by rlambert27** [comment]
I agree with @LaurentGoderre that this does not fail WCAG even though it fails this specific test. The W3C even has an example of this in their tutorial and specifically references a search box:

https://www.w3.org/WAI/tutorials/forms/labels/

However I don't agree with the title attribute. The title sufficient technique contains a warning about using it because it can't be depended upon as not all AT software support and/or interpret the title attribute. The title attribute often contains "supplementary" information and is ignored in most cases.

Personally I wouldn't make any changes to this search box, it's been that way for years with no complaints that I know of.

**Comment by ciclerou** [comment]
Great!  Thanks for the example of where it is used.  I have asked our QA team to confirm.

**Comment by pjackson28** [comment]
Note that it passes the related sufficient technique [G167: Using an adjacent button to label the purpose of a field](http://www.w3.org/TR/WCAG20-TECHS/G167.html) to cover off on the visual labelling.

**Comment by LaurentGoderre** [comment]
Ah there you go. It's been a while since I looked at the techniques

**Comment by ciclerou** [comment]
...and G167 passes :)

**Comment by LaurentGoderre** [comment]
Can we close this then?

**Comment by ciclerou** [comment]
Yes, I will let QA team know.  Thanks
